File: api/audio-book/src/main/java/org/backend/common/response/ApiResponse.java
package org.backend.common.response;


import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

/**
 * Standard generic structure for all HTTP responses in the application.
 *
 * @param <T> the data payload type
 */
@Builder
@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {

    /**
     * Application internal status or HTTP response code.
     */
    @Builder.Default
    private int code=1000;

    /**
     * Human-readable explanation of response status.
     */
    @Builder.Default
    private String message= "success";

    /**
     * Optional body/payload containing result data.
     */
    private T data;
}

File: mobile_client/lib/src/auth/models/api_response.dart
/// Đối tượng phản hồi chung từ API (Generic API Response wrapper).
///
/// Lớp này bọc toàn bộ dữ liệu trả về từ hệ thống API của Backend, bao gồm mã trạng thái, dữ liệu và thông điệp.
class ApiResponse<T> {
  /// Mã phản hồi của API (Ví dụ: 1000 cho thành công).
  final int code;

  /// Dữ liệu phản hồi chính thuộc kiểu generic [T], có thể null.
  final T? data;

  /// Thông điệp phản hồi từ API (ví dụ: 'success' hoặc thông báo lỗi).
  final String message;

  /// Khởi tạo [ApiResponse] với các giá trị mặc định.
  const ApiResponse({
    this.code = 1000,
    this.data,
    this.message = 'success',
  });
}
